gpjax/decision_making/test_functions/non_conjugate_functions.py

```diff
@@ -83,8 +82,8 @@ def evaluate(self, x: Float[Array, "N 1"]) -> Integer[Array, "N 1"]:
             x (Float[Array, 'N D']): Points to evaluate the test function at.
 
         Returns:
-            Integer[Array, 'N 1']: Values of the test function at the points.
+            Float[Array, 'N 1']: Values of the test function at the points.
         """
         key = jr.key(42)
         f = lambda x: 2.0 * jnp.sin(3 * x) + 0.5 * x
-        return jr.poisson(key, jnp.exp(f(x)))
+        return jnp.float64(jr.poisson(key, jnp.exp(f(x))))
```

Comment:
Must this be float64? If the user is using float32 arrays (ill-advised, but possible), then this will cause mixed precision errors. Can we perhaps wrap in a `jnp.asarray()`?

Reply:
Have switched back to returning an array of integers, as these are samples from a Poisson distribution.